src/wallet/sqlite.cpp

```diff
@@ -139,6 +156,8 @@ void SQLiteDatabase::Close()
         throw std::runtime_error(strprintf("SQLiteDatabase: Failed to close database: %s\n", sqlite3_errstr(res)));
     }
     m_db = nullptr;
+
+    UnlockDirectory(m_dir_path, ".walletlock");
```

Comment:
I've removed `if (!m_db) return;` so this will always be run. This is safe because `sqlite3_close` on a nullptr is a no-op. This will unconditionally unlock the directory on db close.